### PR TITLE
Improve OAuth perf and UX

### PR DESCRIPTION
**Pull request by byrichardpowell**
See: https://github.com/Shopify/shopify-app-template-node/pull/947

### Testing this PR

Install flow:

1. Ensure the app is not already installed, and delete `/qrcode/node/web/database.sqlite`
2. Run yarn dev
3. Click the link the CLI outputs
4. Your app should be installed

Missing token flow:

1. Install your app 
2. cd `qr-code/node/web` 
3. `sqlite3 database.sqlite`
4. `DELETE from shopify_sessions;`
5. Refresh the page inside admin that is rendering your app

### CODE REVIEW

File: qr-code/node/web/index.js [hunk 1]
```diff
@@ -122,29 +122,36 @@ export async function createServer(
...
-        .set("Content-Type", "text/html")
-        .send(fs.readFileSync(fallbackFile));
-    }
-  });
+   app.use("/*", async (req, res, next) => {
+
+     if (typeof req.query.shop !== "string") {
+       res.status(500);
+       return res.send("No shop provided");
+     }
```
Comment (paulomarg) [inline qr-code/node/web/index.js]: Just that nit here that we might want to check for this within `sanitizeShop`

File: qr-code/node/web/middleware/auth.js [hunk 2]
```diff
@@ -75,12 +29,12 @@ export default function applyAuthMiddleware(app) {
         }
       });
 
+      const redirectUrl = Shopify.Context.IS_EMBEDDED_APP
+        ? Shopify.Utils.getEmbeddedAppUrl(req)
+        : `/?shop=${session.shop}&host=${encodeURIComponent(req.query.host)}`;
```
Comment (paulomarg) [inline qr-code/node/web/middleware/auth.js:34]: Shouldn't we still be sanitizing the host here?

Comment (byrichardpowell) [reply qr-code/node/web/middleware/auth.js]: Yeah, good call.  Updated 👍

File: qr-code/node/web/index.js (hunk 1 above)
Comment (paulomarg) [reply qr-code/node/web/index.js]: I'll create a task for it.

**Pull request merged**